### PR TITLE
Revert locks

### PR DESCRIPTION
**Pull request by ColinKYuen**
### Summary

Revert locks

### Description

Reverts changes of #58 

### Additional Reviewers

@karenc-bq @matthewh-BQ @sergiyv-bitquill

### CODE REVIEW

File: src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/ClusterAwareConnectionProxy.java [hunk 1]
```diff
@@ -360,7 +359,7 @@ protected void initProxy(ConnectionUrl connUrl) throws SQLException {
     this.initProxy(connUrl, ConnectionPluginManager::new);
   }
 
-  private void initProxy(
+  private synchronized void initProxy(
```
Comment (karenc-bq) [inline src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/ClusterAwareConnectionProxy.java:362]: Are we adding more synchronized keyword in this PR? Was this keyword here in the old code?

File: src/main/user-impl/java/com/mysql/cj/jdbc/ha/MultiHostConnectionProxy.java [hunk 2]
```diff
@@ -330,18 +326,17 @@ protected void invalidateCurrentConnection() throws SQLException {
...
      */
-    protected void invalidateConnection(JdbcConnection conn) throws SQLException {
-        synchronized (this.lockObject) {
-            try {
-                if (conn != null && !conn.isClosed()) {
-                    conn.realClose(true, !conn.getAutoCommit(), true, null);
-                }
-            } catch (SQLException e) {
-                // swallow this exception, current connection should be useless anyway.
+    protected synchronized void invalidateConnection(JdbcConnection conn) throws SQLException {
```
Comment (sergiyv-bitquill) [inline src/main/user-impl/java/com/mysql/cj/jdbc/ha/MultiHostConnectionProxy.java:329]: We can remove synchronized here. InvalidateConnection() has synchronized so no need to do it twice.

File: src/main/user-impl/java/com/mysql/cj/jdbc/ha/MultiHostConnectionProxy.java [hunk 3]
```diff
@@ -316,10 +314,8 @@ boolean isMasterConnection() {
      * @throws SQLException
      *             if an error occurs
      */
-    protected void invalidateCurrentConnection() throws SQLException {
-        synchronized (this.lockObject) {
-            invalidateConnection(this.currentConnection);
-        }
+    protected synchronized void invalidateCurrentConnection() throws SQLException {
```
Comment (sergiyv-bitquill) [inline src/main/user-impl/java/com/mysql/cj/jdbc/ha/MultiHostConnectionProxy.java:317]: We can remove synchronized here. InvalidateConnection() has synchronized so no need to do it twice.

File: src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/ClusterAwareConnectionProxy.java (hunk 1 above)
Comment (sergiyv-bitquill) [reply src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/ClusterAwareConnectionProxy.java:362]: Agree. It's a private function, no need for synchronized here.

File: src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/ClusterAwareConnectionProxy.java [hunk 4]
```diff
@@ -521,7 +520,7 @@ private void setClusterId(String host, int port) {
     }
   }
 
-  private HostInfo createClusterInstanceTemplate(
+  private synchronized HostInfo createClusterInstanceTemplate(
```
Comment (sergiyv-bitquill) [inline src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/ClusterAwareConnectionProxy.java:523]: Don't need it here.

File: src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/ClusterAwareConnectionProxy.java [hunk 5]
```diff
@@ -543,36 +542,34 @@ private HostInfo createClusterInstanceTemplate(
         properties);
   }
 
-  protected void createConnectionAndInitializeTopology(ConnectionUrl connUrl)
+  protected synchronized void createConnectionAndInitializeTopology(ConnectionUrl connUrl)
```
Comment (sergiyv-bitquill) [inline src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/ClusterAwareConnectionProxy.java:545]: Don't need it here.

File: src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/ClusterAwareConnectionProxy.java [hunk 6]
```diff
@@ -543,36 +542,34 @@ private HostInfo createClusterInstanceTemplate(
...
+      ((NativeSession) this.currentConnection.getSession()).setSocketTimeout(this.failoverSocketTimeoutMs);
     }
   }
 
   private boolean isExplicitlyReadOnly() {
     return this.explicitlyReadOnly != null && this.explicitlyReadOnly;
   }
 
-  private void createInitialConnection(ConnectionUrl connUrl) throws SQLException {
+  private synchronized void createInitialConnection(ConnectionUrl connUrl) throws SQLException {
```
Comment (sergiyv-bitquill) [inline src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/ClusterAwareConnectionProxy.java:572]: Don't need it here.

File: src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/ClusterAwareConnectionProxy.java [hunk 7]
```diff
@@ -680,7 +677,7 @@ private int getRandomReaderIndex() {
     return (int) (Math.random() * ((max - min) + 1)) + min;
   }
 
-  private void initTopology() {
+  private synchronized void initTopology() {
```
Comment (sergiyv-bitquill) [inline src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/ClusterAwareConnectionProxy.java:680]: Don't need it here.

File: src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/ClusterAwareConnectionProxy.java [hunk 8]
```diff
@@ -701,7 +698,7 @@ private void initTopology() {
     }
   }
 
-  private void validateInitialConnection() throws SQLException {
+  private synchronized void validateInitialConnection() throws SQLException {
```
Comment (sergiyv-bitquill) [inline src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/ClusterAwareConnectionProxy.java:701]: Don't need it here.

File: src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/ClusterAwareConnectionProxy.java [hunk 9]
```diff
@@ -846,7 +839,7 @@ private boolean isWriterHostIndex(int hostIndex) {
    * @param connection The connection instance to switch to.
    * @throws SQLException if an error occurs
    */
-  private void switchCurrentConnectionTo(int hostIndex, JdbcConnection connection)
+  private synchronized void switchCurrentConnectionTo(int hostIndex, JdbcConnection connection)
```
Comment (sergiyv-bitquill) [inline src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/ClusterAwareConnectionProxy.java:842]: Don't need it here.

File: src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/ClusterAwareConnectionProxy.java [hunk 10]
```diff
@@ -873,7 +866,7 @@ private void switchCurrentConnectionTo(int hostIndex, JdbcConnection connection)
    * @return The new connection instance.
    * @throws SQLException if an error occurs
    */
-  private ConnectionImpl createConnectionForHostIndex(int hostIndex)
+  private synchronized ConnectionImpl createConnectionForHostIndex(int hostIndex)
```
Comment (sergiyv-bitquill) [inline src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/ClusterAwareConnectionProxy.java:869]: Don't need it here.

File: src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/ClusterAwareConnectionProxy.java [hunk 11]
```diff
@@ -886,7 +879,7 @@ private ConnectionImpl createConnectionForHostIndex(int hostIndex)
    * @throws SQLException if an error occurs
    */
   @Override
-  protected ConnectionImpl createConnectionForHost(HostInfo baseHostInfo)
+  protected synchronized ConnectionImpl createConnectionForHost(HostInfo baseHostInfo)
```
Comment (sergiyv-bitquill) [inline src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/ClusterAwareConnectionProxy.java:882]: Don't need it here.

File: src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/ClusterAwareConnectionProxy.java [hunk 12]
```diff
@@ -976,7 +967,7 @@ protected void failoverWriter() throws SQLException {
             new Object[] {this.hosts.get(this.currentHostIndex)}));
   }
 
-  private void processFailoverFailure(String message) throws SQLException {
+  private synchronized void processFailoverFailure(String message) throws SQLException {
```
Comment (sergiyv-bitquill) [inline src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/ClusterAwareConnectionProxy.java:970]: Don't need it here.

File: src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/ClusterAwareConnectionProxy.java [hunk 13]
```diff
@@ -1204,7 +1193,7 @@ protected Object invokeMore(Object proxy, Method method, Object[] args)
     return result;
   }
 
-  private void invalidInvocationOnClosedConnection() throws SQLException {
+  private synchronized void invalidInvocationOnClosedConnection() throws SQLException {
```
Comment (sergiyv-bitquill) [inline src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/ClusterAwareConnectionProxy.java:1196]: Don't need it here.

File: src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/ClusterAwareConnectionProxy.java [hunk 14]
```diff
@@ -1249,7 +1238,7 @@ protected void dealWithIllegalStateException(IllegalStateException e) throws Thr
     dealWithOriginalException(e.getCause(), e);
   }
 
-  private void dealWithOriginalException(
+  private synchronized void dealWithOriginalException(
```
Comment (sergiyv-bitquill) [inline src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/ClusterAwareConnectionProxy.java:1241]: Don't need it here.

File: src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/ClusterAwareConnectionProxy.java [hunk 15]
```diff
@@ -1314,7 +1303,7 @@ protected boolean shouldExceptionTriggerConnectionSwitch(Throwable t) {
    * @throws SQLException if an error occurs
    */
   @Override
-  protected void invalidateCurrentConnection() throws SQLException {
+  protected synchronized void invalidateCurrentConnection() throws SQLException {
```
Comment (sergiyv-bitquill) [inline src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/ClusterAwareConnectionProxy.java:1306]: Don't need it here.

**Review by sergiyv-bitquill — APPROVED** [review]

**Pull request merged**